src/partition/repart.c

```diff
@@ -2492,6 +2492,33 @@ static int context_read_definitions(Context *context) {
 
         }
 
+        /* Make sure we won't end up with more than one ESP or XBOOTLDR partition (which could break the
+         * bootloader) */
+
+        Partition *esp = NULL, *xbootldr = NULL;
+        LIST_FOREACH(partitions, p, context->partitions) {
+                Partition **dest = NULL;
+                const char *desc = NULL;
+                switch (p->type.designator) {
+                case PARTITION_DESIGNATOR_ESP:
+                        dest = &esp;
+                        desc = "EFI System";
+                        break;
+                case PARTITION_DESIGNATOR_XBOOTLDR:
+                        dest = &xbootldr;
+                        desc = "Extended Boot Loader";
+                        break;
+                default:
+                        continue;
+                }
+
+                if (*dest)
+                        return log_error_errno(SYNTHETIC_ERRNO(EINVAL),
+                                               "Defining multiple %s Partitions is unsupported.",
+                                               desc);
+                *dest = p;
+        }
```

Comment:
please split this out into a separate helper maybe. (in fact, the other loops above should probably be split out too, but maybe let's not make th problem worse)

Reply:
The commit (98cb590041d2fc15e39b89cfb0e8ac61236f4f28) is no longer required by the overarching feature so I'm dropping it